jsonforms-tooling-common/test/generate.test.ts

```diff
@@ -0,0 +1,115 @@
+          }
+          return false;
+        },
+        showOpenDialog: async (param: any) => {
+          if (param.id === 'selectSchema') {
+            selectSchemaWasCalled = true;
+            return [{fsPath: schemaPath}];
+          }
+          return false;
+        },
+        showQuickPick: async (options: any, param: any) => {
+          if (param.id === 'overwrite') {
+            overwriteFileWasCalled = true;
+            return 'Yes';
+          }
+          return false;
+        },
+      },
+      uiSchemaName,
+    };
+    return newEditorInstance;
+  };
+
+  test('Check if generated uischema matches correct uischema', async () => {
+    const schemaPath = join(basePath, 'schema.json');
+    const newEditorInstance = getEditor(schemaPath);
+    await generateUISchema(newEditorInstance, schemaPath);
+
+    const testFile = await readFileWithPromise(join(basePath, newEditorInstance.uiSchemaName), 'utf8');
+    const actualFile = await readFileWithPromise(join(basePath, 'uischema.json'), 'utf8');
+    expect(testFile).toBe(actualFile);
+
+    await unlinkWithPromise(join(__dirname, 'assets', newEditorInstance.uiSchemaName));
+  });
+
+  test('Check if vscode inputs are shown correctly', async () => {
+    const schemaPath = join(basePath, 'schema.json');
+    const newEditorInstance = getEditor(schemaPath);
+    await generateUISchema(newEditorInstance, schemaPath);
+
+    expect(selectSchemaWasCalled).toBe(false);
+    expect(enterFileNameWasCalled).toBe(true);
+    expect(overwriteFileWasCalled).toBe(false);
+
+    await unlinkWithPromise(join(__dirname, 'assets', newEditorInstance.uiSchemaName));
+  });
+
+  test('Check if overwrite file is asked', async () => {
+    const schemaPath = join(basePath, 'schema.json');
+    const newEditorInstance = getEditor(schemaPath);
+    await generateUISchema(newEditorInstance, schemaPath);
+    await generateUISchema(newEditorInstance, schemaPath);
+
+    expect(selectSchemaWasCalled).toBe(false);
+    expect(enterFileNameWasCalled).toBe(true);
+    expect(overwriteFileWasCalled).toBe(true);
+
+    await unlinkWithPromise(join(__dirname, 'assets', newEditorInstance.uiSchemaName));
+  });
+
+  test('Check if path is asked from the user, when not provided', async () => {
+    const schemaPath = join(basePath, 'schema.json');
+    const newEditorInstance = getEditor(schemaPath);
+    await generateUISchema(newEditorInstance, '');
+
+    expect(selectSchemaWasCalled).toBe(true);
+    expect(enterFileNameWasCalled).toBe(true);
+    expect(overwriteFileWasCalled).toBe(false);
+
+    await unlinkWithPromise(join(__dirname, 'assets', newEditorInstance.uiSchemaName));
+  });
+
+  test('Check if an error is thrown, when file is uischema', async () => {
+    const schemaPath = join(basePath, 'uischema.json');
+    const newEditorInstance = getEditor(schemaPath);
+    const result = await generateUISchema(newEditorInstance, schemaPath);
+
+    expect(selectSchemaWasCalled).toBe(false);
```

Comment:
Add/update test where `selecteSchemaWasCalled` is true.

Reply:
See https://github.com/TheZoker/jsonforms-tooling/blob/feat/20-add-tests/jsonforms-tooling-common/test/generate.test.ts#L97